### ISSUE TITLE
Samplesheets: Enable remote sync for sample sheets

### ISSUE THREAD

**Issue by mikkonie** [bihealth/sodar-server#959]
Tl;dr: We need the possibility to sync sample sheets from SODAR instance X to SODAR instance Y. There are plans to implement app data mirroring via source/target sites (see undefined##undefined), but that requires a larger effort and should be done after the major breaking changes to SODAR Core v0.10. Also, the use case we have for this specifically *doesn't* use mirrored project memberships from another site. Hence we are going with a more simple solution for now.

## Spec

NOTE: This is still pretty high level so feel free to discuss me about vague/illogical aspects.

- Add projectroles app settings for remote sheet sync in the `samplesheets` app
    * `sheet_sync_enable`: Enable sheet synchronization (bool)
    * `sheet_sync_url`: Full URL for the project on a remote site (string)
    * `sheet_sync_token`: System user access token for sheet sync (string)
- Add celery task to periodically retrieve sheets from another SODAR instance if `sheet_sync_enable==True`
    * Call `SampleSheetISAExportAPIView` on the other "source" SODAR instance
        + You'll need to have a system user with a Knox access token on the instance
        + The user must have at least `project guest` access to the project in question
    * IF sheets have changed since previous sync
        + THEN Import and replace current project sheets with sheet retrieved from the other SODAR API
        + ELSE skip sync
- Modify UI and views in the "target" instance doing the sync  
    * Disallow sheet replacing, editing and deletion if `sheet_sync_enable==True`
        + Even disallowed for superusers!
        + Just read-only actions allowed for sheets..
        + ..with the exception for landing zone uploads and cache updates
        + **TBD:** UCSC ticket creation still allowed?
        + Sheet version list and restoring not available
    * Disable/hide disallowed actions in UI
    * Disallow calling related UI/REST/Ajax views
- Notes
    * For aforementioned reasons, this should NOT utilize the "source/target" remote site stuff at all, for now at least
    * For testing the sheet retrieval, it's ok just to set up a second project on the same site and perform the REST API call to the same server.

## Tasks

- [x] Add project settings for remote sheet sync
- [x] Create logic for remote sheet sync
- [x] Set up remote sheet sync as periodic celery task
- [x] Update views to disallow sheet editing operations if sync is active
- [x] Add tests

## Old Discussion

There is a use case which requires mirroring of sample sheets from one SODAR instance into another in order to be visible on a DMZ instance. It's one-way sync with no edits returned back to the source server.

There are basically two way to do this:

1. Develop the planned remote sync/access framework for app data in SODAR Core (see undefined##undefined) and set up the second SODAR instance as a target site
2. Add a feature into the samplesheets app for syncing the project sample sheet periodically from another SODAR instance, while preventing local import/editing/restoring.

The second option, while kind of a temporary hack, would most likely be **way** faster to design and implement, while essentially providing the same functionality for the user. We already have the sheet export and import APIs in place so half of the work is technically done.

So right now my thinking is this:
- If we need the sample sheet mirroring functionality done fast AND can be fairly certain that it is the only use case we expect for a decent period of time, go with option 2 for now.
- If we have ideas for other use cases for data sync between two SODAR instances and can spare the time, should go with the more elaborate framework of option 1.

**Comment by mikkonie** [comment]
unassigned @stolpeo
